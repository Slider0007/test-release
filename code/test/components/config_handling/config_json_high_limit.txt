Modify JSON elements accordingly to test lower limits and stringfy as compact version: https://jsonformatter.org/json-stringify-online

{
  "config": {
    "version": 3,
    "lastmodified": ""
  },
  "operationmode": {
    "opmode": 2,
    "automaticprocessinterval": "1.0",
    "usedemoimages": false
  },
  "takeimage": {
    "flashlight": {
      "flashtime": 2000,
      "flashintensity": 101
    },
    "camera": {
      "camerafrequency": 21,
      "imagequality": 64,
      "imagesize": "VGA",
      "brightness": 3,
      "contrast": 3,
      "saturation": 3,
      "sharpness": 4,
      "exposurecontrolmode": 2,
      "autoexposurelevel": 5,
      "manualexposurevalue": 1300,
      "gaincontrolmode": 2,
      "manualgainvalue": 1500,
      "specialeffect": 10,
      "mirrorimage": false,
      "flipimage": false,
      "zoommode": 4,
      "zoomoffsetx": 1000,
      "zoomoffsety": 1000
    },
    "debug": {
      "saverawimages": false,
      "rawimageslocation": "/log/source",
      "rawimagesretention": 3
    }
  },
  "imagealignment": {
    "alignmentalgo": 5,
    "searchfield": {
      "x": 20,
      "y": 20
    },
    "imagerotation": "189.9",
    "flipimagesize": false,
    "marker": [
      {
        "x": 1,
        "y": 1
      },
      {
        "x": 1,
        "y": 1
      }
    ],
    "debug": {
      "savedebuginfo": false
    }
  },
  "numbersequences": {
    "sequence": [
      {
        "sequenceid": 0,
        "sequencename": "main"
      }
    ]
  },
  "digit": {
    "enabled": true,
    "model": "dig-class100_0168_s2_q.tflite",
    "cnngoodthreshold": "1.10",
    "sequence": [
      {
        "sequenceid": 0,
        "sequencename": "main",
        "roi": []
      }
    ],
    "debug": {
      "saveroiimages": false,
      "roiimageslocation": "/log/digit",
      "roiimagesretention": 3
    }
  },
  "analog": {
    "enabled": true,
    "model": "ana-class100_0171_s1_q.tflite",
    "sequence": [
      {
        "sequenceid": 0,
        "sequencename": "main",
        "roi": []
      }
    ],
    "debug": {
      "saveroiimages": false,
      "roiimageslocation": "/log/analog",
      "roiimagesretention": 3
    }
  },
  "postprocessing": {
    "sequence": [
      {
        "sequenceid": 0,
        "sequencename": "main",
        "decimalshift": 10,
        "analogdigitsyncvalue": "10.0",
        "extendedresolution": true,
        "ignoreleadingnan": false,
        "checkdigitincreaseconsistency": false,
        "maxratechecktype": 4,
        "maxrate": "0.150",
        "allownegativerate": false,
        "usefallbackvalue": true,
        "fallbackvalueagestartup": 720
      }
    ],
    "debug": {
      "savedebuginfo": false
    }
  },
  "mqtt": {
    "enabled": false,
    "uri": "",
    "maintopic": "watermeter",
    "clientid": "watermeter",
    "authmode": 3,
    "username": "",
    "password": "",
    "tls": {
      "cacert": "",
      "clientcert": "",
      "clientkey": ""
    },
    "processdatanotation": 3,
    "retainprocessdata": false,
    "homeassistant": {
      "discoveryenabled": false,
      "discoveryprefix": "homeassistant",
      "statustopic": "homeassistant/status",
      "metertype": 12,
      "retaindiscovery": false
    }
  },
  "influxdbv1": {
    "enabled": false,
    "uri": "",
    "database": "",
    "authmode": 3,
    "username": "",
    "password": "",
    "tls": {
      "cacert": "",
      "clientcert": "",
      "clientkey": ""
    },
    "sequence": [
      {
        "sequenceid": 0,
        "sequencename": "main",
        "measurementname": "",
        "fieldkey1": ""
      }
    ]
  },
  "influxdbv2": {
    "enabled": false,
    "uri": "",
    "bucket": "",
    "organization": "",
    "authmode": 3,
    "token": "",
    "tls": {
      "cacert": "",
      "clientcert": "",
      "clientkey": ""
    },
    "sequence": [
      {
        "sequenceid": 0,
        "sequencename": "main",
        "measurementname": "",
        "fieldkey1": ""
      }
    ]
  },
  "gpio": {
    "customizationenabled": false,
    "gpiopin": [
      {
        "gpionumber": 1,
        "gpiousage": "restricted: uart0-tx",
        "pinenabled": false,
        "pinname": "",
        "pinmode": "input",
        "capturemode": "cyclic-polling",
        "inputdebouncetime": 5001,
        "pwmfrequency": 1000001,
        "exposetomqtt": false,
        "exposetorest": false,
        "smartled": {
          "type": 8,
          "quantity": 0,
          "colorredchannel": 256,
          "colorgreenchannel": 256,
          "colorbluechannel": 256
        },
        "intensitycorrectionfactor": 101
      },
      {
        "gpionumber": 3,
        "gpiousage": "restricted: uart0-rx",
        "pinenabled": false,
        "pinname": "",
        "pinmode": "input",
        "capturemode": "cyclic-polling",
        "inputdebouncetime": 200,
        "pwmfrequency": 5000,
        "exposetomqtt": false,
        "exposetorest": false,
        "smartled": {
          "type": 0,
          "quantity": 1,
          "colorredchannel": 255,
          "colorgreenchannel": 255,
          "colorbluechannel": 255
        },
        "intensitycorrectionfactor": 100
      },
      {
        "gpionumber": 4,
        "gpiousage": "flashlight-pwm",
        "pinenabled": false,
        "pinname": "",
        "pinmode": "flashlight-default",
        "capturemode": "cyclic-polling",
        "inputdebouncetime": 200,
        "pwmfrequency": 5000,
        "exposetomqtt": false,
        "exposetorest": false,
        "smartled": {
          "type": 0,
          "quantity": 1,
          "colorredchannel": 255,
          "colorgreenchannel": 255,
          "colorbluechannel": 255
        },
        "intensitycorrectionfactor": 100
      },
      {
        "gpionumber": 12,
        "gpiousage": "spare",
        "pinenabled": false,
        "pinname": "",
        "pinmode": "input",
        "capturemode": "cyclic-polling",
        "inputdebouncetime": 200,
        "pwmfrequency": 5000,
        "exposetomqtt": false,
        "exposetorest": false,
        "smartled": {
          "type": 0,
          "quantity": 1,
          "colorredchannel": 255,
          "colorgreenchannel": 255,
          "colorbluechannel": 255
        },
        "intensitycorrectionfactor": 100
      },
      {
        "gpionumber": 13,
        "gpiousage": "spare",
        "pinenabled": false,
        "pinname": "",
        "pinmode": "input",
        "capturemode": "cyclic-polling",
        "inputdebouncetime": 200,
        "pwmfrequency": 5000,
        "exposetomqtt": false,
        "exposetorest": false,
        "smartled": {
          "type": 0,
          "quantity": 1,
          "colorredchannel": 255,
          "colorgreenchannel": 255,
          "colorbluechannel": 255
        },
        "intensitycorrectionfactor": 100
      }
    ]
  },
  "log": {
    "debug": {
      "loglevel": 5,
      "logfilesretention": 5,
      "debugfilesretention": 5
    },
    "data": {
      "enabled": false,
      "datafilesretention": 30
    }
  },
  "network": {
    "wlan": {
      "ssid": "",
      "password": "",
      "hostname": "watermeter",
      "ipv4": {
        "networkconfig": 2,
        "ipaddress": "",
        "subnetmask": "",
        "gatewayaddress": "",
        "dnsserver": ""
      },
      "wlanroaming": {
        "enabled": false,
        "rssithreshold": 1
      }
    },
    "time": {
      "timezone": "CET-1CEST,M3.5.0,M10.5.0/3",
      "ntp": {
        "timesyncenabled": true,
        "timeserver": "",
        "processstartinterlock": true
      }
    }
  },
  "system": {
    "cpufrequency": 241
  },
  "webui": {
    "autorefresh": {
      "overviewpage": {
        "enabled": true,
        "refreshtime": 5
      },
      "datagraphpage": {
        "enabled": false,
        "refreshtime": 60
      }
    }
  }
}