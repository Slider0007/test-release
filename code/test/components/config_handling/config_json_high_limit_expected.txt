Modify JSON elements accordingly to test lower limits and stringfy as compact version: https://jsonformatter.org/json-stringify-online

{
  "config": {
    "version": 3,
    "lastmodified": ""
  },
  "operationmode": {
    "opmode": 1,
    "automaticprocessinterval": "1.0",
    "usedemoimages": false
  },
  "takeimage": {
    "flashlight": {
      "flashtime": 2000,
      "flashintensity": 100
    },
    "camera": {
      "camerafrequency": 20,
      "imagequality": 63,
      "imagesize": "VGA",
      "brightness": 2,
      "contrast": 2,
      "saturation": 2,
      "sharpness": 3,
      "exposurecontrolmode": 2,
      "autoexposurelevel": 2,
      "manualexposurevalue": 1200,
      "gaincontrolmode": 1,
      "manualgainvalue": 5,
      "specialeffect": 7,
      "mirrorimage": false,
      "flipimage": false,
      "zoommode": 2,
      "zoomoffsetx": 960,
      "zoomoffsety": 720
    },
    "debug": {
      "saverawimages": false,
      "rawimageslocation": "/log/source",
      "rawimagesretention": 3
    }
  },
  "imagealignment": {
    "alignmentalgo": 4,
    "searchfield": {
      "x": 20,
      "y": 20
    },
    "imagerotation": "180.0",
    "flipimagesize": false,
    "marker": [
      {
        "x": 1,
        "y": 1
      },
      {
        "x": 1,
        "y": 1
      }
    ],
    "debug": {
      "savedebuginfo": false
    }
  },
  "numbersequences": {
    "sequence": [
      {
        "sequenceid": 0,
        "sequencename": "main"
      }
    ]
  },
  "digit": {
    "enabled": true,
    "model": "dig-class100_0168_s2_q.tflite",
    "cnngoodthreshold": "1.00",
    "sequence": [
      {
        "sequenceid": 0,
        "sequencename": "main",
        "roi": []
      }
    ],
    "debug": {
      "saveroiimages": false,
      "roiimageslocation": "/log/digit",
      "roiimagesretention": 3
    }
  },
  "analog": {
    "enabled": true,
    "model": "ana-class100_0171_s1_q.tflite",
    "sequence": [
      {
        "sequenceid": 0,
        "sequencename": "main",
        "roi": []
      }
    ],
    "debug": {
      "saveroiimages": false,
      "roiimageslocation": "/log/analog",
      "roiimagesretention": 3
    }
  },
  "postprocessing": {
    "sequence": [
      {
        "sequenceid": 0,
        "sequencename": "main",
        "decimalshift": 9,
        "analogdigitsyncvalue": "9.9",
        "extendedresolution": true,
        "ignoreleadingnan": false,
        "checkdigitincreaseconsistency": false,
        "maxratechecktype": 2,
        "maxrate": "0.150",
        "allownegativerate": false,
        "usefallbackvalue": true,
        "fallbackvalueagestartup": 720
      }
    ],
    "debug": {
      "savedebuginfo": false
    }
  },
  "mqtt": {
    "enabled": false,
    "uri": "",
    "maintopic": "watermeter",
    "clientid": "watermeter",
    "authmode": 2,
    "username": "",
    "password": "",
    "tls": {
      "cacert": "",
      "clientcert": "",
      "clientkey": ""
    },
    "processdatanotation": 2,
    "retainprocessdata": false,
    "homeassistant": {
      "discoveryenabled": false,
      "discoveryprefix": "homeassistant",
      "statustopic": "homeassistant/status",
      "metertype": 10,
      "retaindiscovery": false
    }
  },
  "influxdbv1": {
    "enabled": false,
    "uri": "",
    "database": "",
    "authmode": 2,
    "username": "",
    "password": "",
    "tls": {
      "cacert": "",
      "clientcert": "",
      "clientkey": ""
    },
    "sequence": [
      {
        "sequenceid": 0,
        "sequencename": "main",
        "measurementname": "",
        "fieldkey1": ""
      }
    ]
  },
  "influxdbv2": {
    "enabled": false,
    "uri": "",
    "bucket": "",
    "organization": "",
    "authmode": 2,
    "token": "",
    "tls": {
      "cacert": "",
      "clientcert": "",
      "clientkey": ""
    },
    "sequence": [
      {
        "sequenceid": 0,
        "sequencename": "main",
        "measurementname": "",
        "fieldkey1": ""
      }
    ]
  },
  "gpio": {
    "customizationenabled": false,
    "gpiopin": [
      {
        "gpionumber": 1,
        "gpiousage": "restricted: uart0-tx",
        "pinenabled": false,
        "pinname": "",
        "pinmode": "input",
        "capturemode": "cyclic-polling",
        "inputdebouncetime": 5000,
        "pwmfrequency": 1000000,
        "exposetomqtt": false,
        "exposetorest": false,
        "smartled": {
          "type": 5,
          "quantity": 1,
          "colorredchannel": 255,
          "colorgreenchannel": 255,
          "colorbluechannel": 255
        },
        "intensitycorrectionfactor": 100
      },
      {
        "gpionumber": 3,
        "gpiousage": "restricted: uart0-rx",
        "pinenabled": false,
        "pinname": "",
        "pinmode": "input",
        "capturemode": "cyclic-polling",
        "inputdebouncetime": 200,
        "pwmfrequency": 5000,
        "exposetomqtt": false,
        "exposetorest": false,
        "smartled": {
          "type": 0,
          "quantity": 1,
          "colorredchannel": 255,
          "colorgreenchannel": 255,
          "colorbluechannel": 255
        },
        "intensitycorrectionfactor": 100
      },
      {
        "gpionumber": 4,
        "gpiousage": "flashlight-pwm",
        "pinenabled": false,
        "pinname": "",
        "pinmode": "flashlight-default",
        "capturemode": "cyclic-polling",
        "inputdebouncetime": 200,
        "pwmfrequency": 5000,
        "exposetomqtt": false,
        "exposetorest": false,
        "smartled": {
          "type": 0,
          "quantity": 1,
          "colorredchannel": 255,
          "colorgreenchannel": 255,
          "colorbluechannel": 255
        },
        "intensitycorrectionfactor": 100
      },
      {
        "gpionumber": 12,
        "gpiousage": "spare",
        "pinenabled": false,
        "pinname": "",
        "pinmode": "input",
        "capturemode": "cyclic-polling",
        "inputdebouncetime": 200,
        "pwmfrequency": 5000,
        "exposetomqtt": false,
        "exposetorest": false,
        "smartled": {
          "type": 0,
          "quantity": 1,
          "colorredchannel": 255,
          "colorgreenchannel": 255,
          "colorbluechannel": 255
        },
        "intensitycorrectionfactor": 100
      },
      {
        "gpionumber": 13,
        "gpiousage": "spare",
        "pinenabled": false,
        "pinname": "",
        "pinmode": "input",
        "capturemode": "cyclic-polling",
        "inputdebouncetime": 200,
        "pwmfrequency": 5000,
        "exposetomqtt": false,
        "exposetorest": false,
        "smartled": {
          "type": 0,
          "quantity": 1,
          "colorredchannel": 255,
          "colorgreenchannel": 255,
          "colorbluechannel": 255
        },
        "intensitycorrectionfactor": 100
      }
    ]
  },
  "log": {
    "debug": {
      "loglevel": 4,
      "logfilesretention": 5,
      "debugfilesretention": 5
    },
    "data": {
      "enabled": false,
      "datafilesretention": 30
    }
  },
  "network": {
    "wlan": {
      "ssid": "",
      "password": "",
      "hostname": "watermeter",
      "ipv4": {
        "networkconfig": 0,
        "ipaddress": "",
        "subnetmask": "",
        "gatewayaddress": "",
        "dnsserver": ""
      },
      "wlanroaming": {
        "enabled": false,
        "rssithreshold": 0
      }
    },
    "time": {
      "timezone": "CET-1CEST,M3.5.0,M10.5.0/3",
      "ntp": {
        "timesyncenabled": true,
        "timeserver": "",
        "processstartinterlock": true
      }
    }
  },
  "system": {
    "cpufrequency": 240
  },
  "webui": {
    "autorefresh": {
      "overviewpage": {
        "enabled": true,
        "refreshtime": 5
      },
      "datagraphpage": {
        "enabled": false,
        "refreshtime": 60
      }
    }
  }
}